addons/google_gmail/tests/test_google_gmail.py

```diff
@@ -0,0 +1,57 @@
+from odoo.tests.common import TransactionCase
+from unittest import mock
+from datetime import datetime
+from freezegun import freeze_time
+
+
+class TestIrMailServer(TransactionCase):
+
+    @freeze_time("2021-12-15 10:59:50")
+    def test_generate_oauth2_string_token_valid(self):
```

Comment:
As most setup is common, can't you extract it into a setup and keep the test itself in test method(s) ?

Reply:
done !